src/compiler/checker.ts

```diff
@@ -69,6 +69,9 @@ namespace ts {
         undefinedSymbol.declarations = [];
         const argumentsSymbol = createSymbol(SymbolFlags.Property, "arguments");
 
+        /** This will be set during calls to `getResolvedSignature` where services determines an apparent number of arguments greater than what is actually provided. */
+        let apparentArgumentCount: number | undefined;

```

Comment:
It would be nice if we just computed this at the moment we need it, but that seems to involve getting the tokens array, which is currently code exclusive to services. See `getArgumentCount` in `signatureHelp.ts`.